pkg/apply/taskrunner/context.go

```diff
@@ -30,3 +34,26 @@ func (tc *TaskContext) TaskChannel() chan TaskResult {
 func (tc *TaskContext) EventChannel() chan event.Event {
 	return tc.eventChannel
 }
+
+// ResourceApplied updates the context with information about the
+// resource identified by the provided id. Currently, we keep information
+// about the generation of the resource after the apply operation completed.
+func (tc *TaskContext) ResourceApplied(id object.ObjMetadata, gen int64) {
+	tc.appliedResources[id] = applyInfo{
+		generation: gen,
+	}
+}
+
+// ResourceGeneration looks up the generation of the given resource
+// after it was applied.
+func (tc *TaskContext) ResourceGeneration(id object.ObjMetadata) int64 {
+	ai, found := tc.appliedResources[id]
+	if !found {
+		return 0
+	}
+	return ai.generation
+}
+
+type applyInfo struct {
```

Comment:
Did you say the `generation` doesn't change when a label is added? Should we comment here about when the `generation` changes?

Reply:
Added comments